### PR TITLE
rework user settings

### PR DESCRIPTION
**Pull request by M1chaCH**
reworked the entire user settings 
- everything works now via autosave
- i think it looks a bit cleaner
- languages aren't saved yet because only one is implemented -> can't test
- changing the password fails unhandled when the old password is wrong. This is because I could not find a clean way to temporarily disable the error handler interceptor. @EricWermelinger any solution ideas?

### CODE REVIEW

File: src/app/Pages/user-settings/user-settings.component.ts [hunk 1]
```diff
@@ -1,95 +1,101 @@
...
-    changeDarkTheme(isDarkTheme: boolean) {
-        this.darkTheme.setDarkTheme(isDarkTheme);
-        const formValue = {
-            ...this.prevUser,
-            darkTheme: isDarkTheme,
-        } as UserOwnChangeDTO;
-        this.prevUser = formValue;
-        this.userService.saveUser$(formValue).subscribe();
+    ngOnDestroy() {
+        this.unsubscribe.next()
```
Comment (EricWermelinger) [inline src/app/Pages/user-settings/user-settings.component.ts:99]: I think to make it work with the takeUntil, we have to add both .next() and .complete()
--> this.unsubscribe.next() to trigger changes
--> this.unsubscribe.complete() to complete the Subject and trigger the take until.

**Review by EricWermelinger — APPROVED** [review]
Looks good, just the one thing with the .next and .copmlete that I would change.

File: src/app/Pages/user-settings/user-settings.component.ts (hunk 1 above)
Comment (M1chaCH) [reply src/app/Pages/user-settings/user-settings.component.ts]: I've changed it, is this what you had in mind? [(changes)](https://github.com/moreach/teachu-client/pull/40/commits/fe8f9bc09e1a9b4d6cf6408c27608c6d036acb00)

Comment (M1chaCH) [reply src/app/Pages/user-settings/user-settings.component.ts]: implemented as discussed via WhatsApp

**Pull request merged**